ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -461,6 +464,10 @@ public void replicate() {
     } finally {
       replicationMetrics.updateOneCycleReplicationTime(time.milliseconds() - oneRoundStartTimeMs,
           replicatingFromRemoteColo, datacenterName);
+      groupFinishTime.forEach((groupId, finishTime) -> {
```

Comment:
Move it out of the finally block and put it right outside the loop. We don't want to measure any exception cases.

Reply:
@snalli @manbearpig1996 I understand the claim from skew perspective. How often we get exceptions? If its close to 0, it should be okay to ignore. If its significant, can we have the exception time as the cycle finish time and still publish the wait time?